ykbh/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+use std::alloc::{alloc, dealloc, Layout};
+use std::convert::{TryFrom, TryInto};
+use ykpack::{
+    self, Constant, ConstantInt, IPlace, Local, LocalDecl, Statement, Terminator, TypeId,
+    UnsignedInt,
+};
+use yktrace::sir::SIR;
+
+const SUSIZE: usize = std::mem::size_of::<usize>();
+
+pub struct SIRInterpreter {
+    map: *mut u8,
```

Comment:
What are `map` and `layout`? I was expecting something like:

```rust
struct SIRInterp {
    locals: *mut u8
}
```

but maybe I've misunderstood what's needed!

Reply:
We need to hold on to the layout so we can deallocate the memory when the interpreter is dropped. Or will recreating the same layout for the `dealloc` call also work?

Reply:
Once the locals are statically calculated, the `Layout` should be trivially calculable (it should become equivalent to a load of a size & alignment) and thus wouldn't need to be stored here.

Reply:
Yeah, I missed that. Will fix.

Reply:
Fixed in b5ac93e.